llvm/unittests/ADT/FunctionExtrasTest.cpp

```diff
@@ -316,7 +316,7 @@ TEST(UniqueFunctionTest, InlineStorageWorks) {
   // We do assume a couple of implementation details of the unique_function here:
   //  - It can store certain small-enough payload inline
   //  - Inline storage size is at least >= sizeof(void*)
-  void *ptr;
+  void *ptr = nullptr;
   unique_function<void(void *)> UniqueFunctionWithInlineStorage{
       [ptr](void *self) {
```

Comment:
In this case we just need to store _something_ in the lambda so we can then get the address of it. Capturing `ptr` by reference here won't really work since we'll be getting the address of the stack variable in this case.

Reply:
Well if we capture by reference we will store the reference to `ptr` as a lambda member.

Reply:
Indeed, but it is impossible to get an address of a reference variable in C++ AFAIK, so we won't be able to use it for the test below? Doing `&ptr` will return the address of original object. Unless I'm misunderstanding what you're suggesting in which case I apologize for not getting that.

Reply:
address of a reference is the underlying pointer

Reply:
> Doing &ptr will return the address of original object.

Yes. Oh I think now I get what this is trying to test -- that the data member is indeed inside the function object. Please disregard then.

Reply:
No worries at all, unfortunately I don't have commit access, so if you could merge this fix for me please that'd be great 👍